public/app/plugins/datasource/prometheus/querybuilder/shared/parsingUtils.ts

```diff
@@ -115,7 +115,7 @@ export function makeBinOp(
  * @param cur
  * @param type
  */
-export function getAllByType(expr: string, cur: SyntaxNode, type: string): string[] {
+export function getAllByType(expr: string, cur: SyntaxNode, type: string | number): string[] {
   if (cur.name === type) {
```

Comment:
This is not going to work as we are passing number (`id`) and comparing it to name. We need to then compare it to the `node.type.id`. Thats why also the test failed.

```suggestion
export function getAllByType(expr: string, cur: SyntaxNode, type: number): string[] {
  if (cur.type.id === type) {
```

Reply:
@ivanahuckova I implemented this change, but since Loki is still calling this function using strings, it broke the build. I updated this to work with either strings from loki or the numbers from prometheus, let me know what you think!